pkg/image/controller/signature/signature_import_controller.go

```diff
@@ -106,8 +106,13 @@ func (s *SignatureImportController) work() bool {
 
 	err := s.syncImageSignatures(key.(string))
 	if err != nil {
-		utilruntime.HandleError(fmt.Errorf("error syncing image %s, it will be retried: %v", key.(string), err))
-		s.queue.AddRateLimited(key)
+		if _, ok := err.(GetSignaturesError); !ok {
+			utilruntime.HandleError(fmt.Errorf("error syncing image %s, it will be retried: %v", key.(string), err))
+		}
+
+		if s.queue.NumRequeues(key) < 5 {
```

Comment:
can you make the 5 a const? `maxSignatureFetchRetries` or something

Reply:
why? we're not using it anywhere else.  if you're not using a constant in two places, there's not much point in making it a constant.

Reply:
I would say there is still point in that: if someone writing a new peace of code will be looking for such constant in the package and the is none he will either create it or use your argument about not using a constant if he uses it only in one place and magic numbers will spread

Reply:
If you have constants they also spread.  Private constants in a package are generally what we try to do though to pull them into one place and put documentation on them.